EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -0,0 +1,466 @@
+//int FEDIDmax = FEDNumbering::MAXHCALuTCAFEDID;
+
+namespace CDFHeaderSpec{
+  static const int OFFSET_H = 0; 
+  static const int MASK_H = 0x8; 
+  static const int OFFSET_FOV = 4;
+  static const int MASK_FOV = 0xF;
+  static const int OFFSET_SOURCE_ID = 8; 
+  static const int MASK_SOURCE_ID = 0xFFF; 
+  static const int OFFSET_BX_ID = 20; 
+  static const int MASK_BX_ID = 0xFFF;
+  static const uint64_t OFFSET_LV1_ID = 32;
+  static const int MASK_LV1_ID = 0xFFFFFF;
+  static const int OFFSET_EVT_TY = 56;
+  static const int MASK_EVT_TY = 0xF;
+  static const int OFFSET_FIXED_MSB = 60; 
+  static const int MASK_FIXED_MSB = 0xF;
+  static const uint64_t FIXED_MSB = 0x5;
+}
+
+namespace AMC13HeaderSpec{
+  static const int OFFSET_FIXED_LSB = 0;
+  static const int MASK_FIXED_LSB = 0xF;
+  static const int FIXED_LSB = 0x0;
+  static const int OFFSET_ORN = 4;
+  static const int MASK_ORN = 0xFFFFFFFF;
+  static const int OFFSET_RESERVED = 36;
+  static const int MASK_RESERVED = 0xFFFF;
+  static const int OFFSET_NAMC = 52;
+  static const int MASK_NAMC = 0xF;
+  static const int OFFSET_RES = 56;
+  static const int MASK_RES = 0xF;
+  static const int OFFSET_UFOV = 60;
+  static const int MASK_UFOV = 0xF;
+}
+
+namespace AMCHeaderSpec{
+  static const int OFFSET_CRATE_ID = 0;
+  static const int MASK_CRATE_ID = 0xFF;
+  static const int OFFSET_SLOT_ID = 8;
+  static const int MASK_SLOT_ID = 0xF;
+  static const int OFFSET_PRESAMPLES = 12;
+  static const int MASK_PRESAMPLES = 0xF;
+  static const int OFFSET_AMC_NO = 16;
+  static const int MASK_AMC_NO = 0xF;
+  static const int OFFSET_BLK_NO = 20;
+  static const int MASK_BLK_NO = 0xFF;
+  static const int OFFSET_FIXED = 28;
+  static const int MASK_FIXED = 0xF;
+  static const int OFFSET_AMCN_SIZE = 32;
+  static const int MASK_AMCN_SIZE = 0xFFFFFF;
+  static const int OFFSET_C = 56;
+  static const int MASK_C = 0x1;
+  static const int OFFSET_V = 57;
+  static const int MASK_V = 0x1;
+  static const int OFFSET_P = 58;
+  static const int MASK_P = 0x1;
+  static const int OFFSET_E = 59;
+  static const int MASK_E = 0x1;
+  static const int OFFSET_S = 60;
+  static const int MASK_S = 0x1;
+  static const int OFFSET_M = 61;
+  static const int MASK_M = 0x1;
+  static const int OFFSET_L = 62;
+  static const int MASK_L = 0x1;    
+}
+
+namespace QIE8HeaderSpec{
+  static const int OFFSET_FIBERCHAN = 0;
+  static const int MASK_FIBERCHAN = 0x3;
+  static const int OFFSET_FIBER = 2;
+  static const int MASK_FIBER = 0x1F;
+  static const int OFFSET_CAPID = 8;
+  static const int MASK_CAPID = 0x3;
+  static const int OFFSET_FIBERERR = 10;
+  static const int MASK_FIBERERR = 0x2;
+  static const int OFFSET_FLAVOER = 12;
+  static const int MASK_FLAVOR = 0x7;
+  static const int OFFSET_HEADER_BIT = 15;
+  static const int MASK_HEADER_BIT = 0x1;
+}
+
+class HCalFED{
+
+public:
+  
+  //FEDMap feds;
+  vector<unsigned char> fedData;
+  vector<uint64_t> AMCHeaders;
```

Comment:
data members should nominally be encapsulated, not exposed as public.
This appears to be a complex enough class that they should be.
Members that are set and accessed verbatim (without extra packing/unpacking manipulations) could probably remain public .. all considering that this lives in plugin/*.h  and isn't exposed to much client code.